scipy/stats/tests/test_mstats_basic.py

```diff
@@ -42,48 +42,88 @@ def test_mquantiles_limit_keyword(self):
         assert_almost_equal(quants, desired)
 
 
-class TestGMean(object):
-    def test_1D(self):
-        a = (1, 2, 3, 4)
-        actual = mstats.gmean(a)
+class MStatsTestMethod(object):
+    def equal_test(self, array_like, desired, axis=None, dtype=None, significant=7):
+        # Note this doesn't test when axis is not specified
+        rtol = np.float_power(10, -1.0 * significant)
+        x = mstats.gmean(array_like, axis=axis, dtype=dtype)
+        assert_allclose(x, desired, rtol=rtol)
+        assert_equal(x.dtype, dtype)
+
+
+class TestGeoMean(MStatsTestMethod):
+    def test_1d(self):
+        a = [1, 2, 3, 4]
         desired = np.power(1*2*3*4, 1./4.)
-        assert_almost_equal(actual, desired, decimal=14)
+        self.equal_test(a, desired, significant=14)
 
         desired1 = mstats.gmean(a, axis=-1)
```

Comment:
Here and in a few places below we are still using ``desired = mstats.gmean``.

Reply:
In this case, the previous version also was using `mstats.gmean()`. 
maybe, in my opinion, this part of `desired1 = mstats.gmean()` seem to be removed.